### PR TITLE
Add MissionClockEvent to track mission clock time.

### PR DESCRIPTION
**Pull request by pmtischler**
For #45 

Following this PR:
1) Refactor [TakeoffOrLandingEvent.flights](https://github.com/auvsi-suas/interop/blob/master/server/auvsi_suas/models/takeoff_or_landing_event.py#L22) so generic implementation it is part of  [TimePeriod](https://github.com/auvsi-suas/interop/blob/master/server/auvsi_suas/models/time_period.py).
2) Refactor [TakeoffOrLandingEvent.user_in_air](https://github.com/auvsi-suas/interop/blob/master/server/auvsi_suas/models/takeoff_or_landing_event.py#L64) so it uses a function in `AccessLog` which gets the latest access log before a time. 
3) Add a method similar to `flights` and `user_in_air` to `MissionClockEvent` in order to get the on-mission-clock periods, on timeout periods, and whether the user is on mission clock / timeout.
4) Add grading of time-on-mission-clock.
5) In future target grading interfaces, hide submitted targets for users which are either on mission clock or on timeout.

### CODE REVIEW

**Comment by pmtischler** [comment]
Forgot to run formatter, doing that now.

File: server/auvsi_suas/models/mission_clock_event.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class MissionClockEvent(AccessLog):
+    """Marker for a team going on/off mission clock."""
+    # Whether the team is now on the mission clock.
+    team_on_clock = models.BooleanField()
+    # Whether the team is now in a timeout.
+    team_on_timeout = models.BooleanField()
+
+    def __unicode__(self):
+        """Descriptive text for use in displays."""
+        return unicode('MissionClockEvent (pk:%s, user:%s, timestamp:%s, '
```
Comment (prattmic) [inline server/auvsi_suas/models/mission_clock_event.py]: No space before ' ('

**Comment by prattmic** [comment]
LGTM

I wonder if this is the best way to approach the problem. All of these edge events are difficult to query and figure out the times when the team is on the clock. I'm not really sure there is a better way to do this, but it seems like what we really want is a list of time ranges when the team is on the clock.

**Comment by pmtischler** [comment]
I thought about that as well. The list of time ranges is an implementation detail of this overall idea, and I think each has their +/-s. If we come up with a clear reason to switch we can always revisit, this part of the code base is small & easy to maintain.

**Pull request merged**